### DISCUSSION TITLE
Wiform MVVM

### DISCUSSION THREAD

**Discussion by dongpo1996** [BYJRK/DotNet-Discussions#382] [MVVM]
在Winfrom中使用Reactive UI实现MVVM架构的情况下ViewModel里面声明了一个BindingList集合直接绑定了前台的DataGridView控件的DataSource属性,ViewModel开了一个新的线程更新BindingList集合数据会出报跨线程访问控件的问题,WPF里面可以直接用Dispatcher来解决,在Winform没有找到合适的解决方案,我现在的方式是直接把Form传到ViewModel,然后调用Form的Invoke方法问题可以得到解决,但是这显然违背了MVVM,不知道大家有没有好的解决方案

**Comment by xkz1994** [comment]
SynchronizationContext可以解决

**Reply by dongpo1996** [reply]
确实可以  谢谢

**Comment by arcanix601** [comment]
ReactiveUI提供了类似`Dispatcher`的 `RxApp.MainThreadScheduler`

**Comment by 158545614** [comment]
wpf里面这样操作好像没有问题
